env.d.ts

```diff
@@ -14,6 +14,24 @@ type ScriptFailure = {
 type ScriptResponse<T = object> = ScriptSuccess<T> | ScriptFailure
 type ErrorScripts = Record<string, () => ScriptFailure>
 
+type AllOptional<T> = {
+	[K in keyof T]-?: {} extends Pick<T, K> ? true : false;
+}[keyof T] extends true ? true : false;
+
+type Scriptor<Args = unknown, Ret = unknown> = {
+	name: string,
+	call: AllOptional<Args> extends true
+		? (args?: Args) => Ret
+		: (args: Args) => Ret;
+};
+
+/*
+type Scriptor<Args = unknown, Ret = ScriptResponse> = {
+	name: string,
+	call: (args?: Args) => Ret
+}
+*/
+
```

Comment:
this looks good but please could you remove the comment and semicolons. after that I think this is good to go.
btw what's the reasoning for `extends true ? true : false`? couldn't that bit be removed and it would still act the same?

Reply:
To be completely honest, that was a ChatGPT consultation, so most likely a hallucination that it needed to be there :P